scripts/eurostat/process_eurostat.py

```diff
@@ -0,0 +1,130 @@
+import pandas as pd
+import argparse
+import gzip
+import shutil
+
+
+def unzip_single(filename_in):
+    filename_out = filename_in[:-3]  # to trim the .gz
+    with gzip.open(filename_in, 'rb') as f_in:
+        with open(filename_out, 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+    return filename_out
+
+
+def first_stage_processing(df, separation_symbol):
+    time_invariant_var = 'geo'
+    header = list(df.columns)
+    variable_column = header[0]
+    variable_column = variable_column.replace('\\', ',')
+    variable_column = variable_column.split(',')
+    assert variable_column[-1] == 'time'  # have this constraint for now
+    assert time_invariant_var in variable_column
+    time_invariant_var_index = variable_column.index(time_invariant_var)
+    time_columns = header[1:]
+    # just strip the space in the time column
+    for i in range(len(time_columns)):
+        time_columns[i] = time_columns[i].strip()
+    df.columns = [header[0]] + time_columns
+    combined_variable_name = ''
+    for v in variable_column:
+        if v != 'time' and v != time_invariant_var:
+            combined_variable_name = combined_variable_name + v + separation_symbol
+    combined_variable_name = combined_variable_name[:-len(separation_symbol)]  # remove last underscore
+    # insert a dummy variable for two dimensional dataset. Note most of the time this should not happen
+    if combined_variable_name == '':
+        combined_variable_name = 'dummy_var_column'
+    total_record_count = len(df)*len(time_columns)
+    all_records = []
+    for i in range(len(df)):
```

Comment:
Do we want to process the header row?  I think so, but I just want to ask to be sure.